### ISSUE TITLE
clique.assemble does not work with collections that do not start with 0

### ISSUE THREAD

**Issue by elveatles** [4degrees/clique#21]
If I try to assemble a list with numbers that do not start with zero, this breaks the collections created. 
For example:

    items = ['file.1001.jpg', 'file.1002.jpg', 'file.1003.jpg']
    collections, remainder = clique.assemble(items)

The collections variable is:
    
    [<Collection "file.%d.jpg [1001-1003]">]

It should be:
    
    [<Collection "file.%04d.jpg [1001-1003]">]

**Comment by martinpengellyphillips** [comment]
Hey there,

The collection you show actually has no padding so Clique is doing the right thing here with the information given. See also #10 

If you want to force a sequence to 4 padded then just manually set the padding attribute on the collection instance:

```python
collection.padding = 4
```

**Comment by elveatles** [comment]
In my situation setting the padding to 4 is fine. However, what if someone needs the padding to match the number of digits from the input?

**Comment by martinpengellyphillips** [comment]
I'm interested in this use case, but don't understand it properly yet. 

Can you explain more about when there is a need for an unpadded sequence to be interpreted as a padded sequence by default?

**Comment by elveatles** [comment]
I think that using hash padding might be a good example since lots of programs use it. For example, converting clique padding to hash padding like this:

    items = ['file.1001.jpg', 'file.1002.jpg', 'file.1003.jpg']
    collections, remainder = clique.assemble(items)
    for collection in collections:
        print collection.format('{head}' + ('#' * collection.padding) + '{tail}')

Currently result is:

    file..jpg

But I would hope for

    file.####.jpg

instead. If I'm trying to force the padding to always be 4, I could just do '#' * 4, but I'd rather this work in a more general case where it doesn't matter what kind of sequence it gets. It would be able to match the padding of the input.

**Comment by martinpengellyphillips** [comment]
To be clear, padding (as understood by Clique) does not simply refer to the width of the maximum index. The reason for this is that it would be incorrect for an unpadded sequence of 1-100 where padding should be 0 not 3. 

In your example, it is ambiguous whether the sequence is padded or not - whether a preceeding index would have been 999 or 0999 - so Clique plays it safe and assumes unpadded.

As this confusion regarding padding has come up a few times though (#10, #15) it might make sense to add a flag to `assemble` to control this decision (something like `assume_padded_when_ambiguous=True`).

**Comment by elveatles** [comment]
Yes, it is ambiguous. I suppose there's no way to know for sure what the padding is if there are no zeros. '1001'.zfill(4) is still '1001', but to know the reverse is impossible.

**Comment by martinpengellyphillips** [comment]
Will add `assume_padded_when_ambiguous` or equivalent flag to control assumptions.
